src/TempleLineOfCredit.sol

```diff
@@ -247,49 +343,95 @@ contract TempleLineOfCredit is Ownable, Operators {
+            
+            if (debtTokenInfo.tokenType == TokenType.TRANSFER){
+                IERC20(debtToken).safeTransferFrom(
+                    msg.sender,
+                    address(this),
+                    repayAmount 
+                );
+            } else {
+                IERC20MintBurn(debtToken).burn(
+                    msg.sender,
+                    repayAmount 
+                );
+            }
+            emit Repay(msg.sender, repayAmount);
         }
-
-        positions[msg.sender].debtAmount -= repayAmount;
-        debtBalance += repayAmount;
-        positions[msg.sender].createdAt = block.timestamp;
-
-        debtToken.safeTransferFrom(
-            msg.sender,
-            address(this),
-            repayAmount 
-        );
-        emit Repay(msg.sender, repayAmount);
     }
    
    /**
      * @dev Allows operator to liquidate debtors position
      * @param debtor the account to liquidate
      */
-    function liquidate(address debtor) external onlyOperators {
-        Position storage position = positions[debtor];    
-        uint256 totalDebtOwed = getTotalDebtAmount(debtor);
+    function liquidate(address debtor, address debtToken) external onlyOperators {
 
-        if (_getCurrentCollaterilizationRatio(position.collateralAmount, position.debtAmount, totalDebtOwed) >= minCollateralizationRatio) {
+        DebtToken memory debtTokenInfo = debtTokens[debtToken];
+        if (!debtTokens[debtToken].isAllowed) revert Unsupported(debtToken);
+        
+        TokenPosition storage userPosition = positions[debtor].tokenPosition[debtToken];
+        uint256 totalDebtOwed = userPosition.debtAmount;
+        uint256 collateralAmount = positions[debtor].collateralAmount;
+        if (totalDebtOwed != 0) {
+                totalDebtOwed = getTotalDebtAmount(debtToken, debtor);
+        }
+
+        if (_getCurrentCollaterilizationRatio(collateralAmount, userPosition.debtAmount, debtTokenInfo.tokenPrice,  totalDebtOwed) >= debtTokenInfo.minCollateralizationRatio) {
             revert OverCollaterilized(debtor);
         }
 
-        uint256 collateralSeized = (totalDebtOwed * debtPrice) / collateralPrice;
+        (uint256 debtTokenPrice, uint256 debtPrecision) = getTokenPrice(debtTokenInfo.tokenPrice);
+        (uint256 collateralTokenPrice, uint256 collateralPrecision) = getTokenPrice(collateralPrice);
+
 
-        if (collateralSeized > position.collateralAmount) {
-            collateralSeized = position.collateralAmount;
+        uint256 collateralSeized = (totalDebtOwed * debtTokenPrice * collateralPrecision) / collateralTokenPrice / debtPrecision;
+
+        if (collateralSeized > collateralAmount) {
+            collateralSeized = collateralAmount;
         }
 
-        position.collateralAmount -= collateralSeized;
-        position.debtAmount = 0;
-        position.createdAt = 0;
+        positions[debtor].collateralAmount -= collateralSeized;
```

Comment:
An unnecessary sload here since it's storage. Better as
```
positions[debtor].collateralAmount = collateralAmount - collateralSeized
```
since you already have it in memory